base_layer/wallet/src/transaction_service/protocols/transaction_send_protocol.rs

```diff
@@ -281,17 +282,44 @@ where
             ));
         }
 
+        // Calculate the size of the transaction - initial send transaction to the peer (always a small message) should
+        // not be attempted if the final transaction size will be too large to be broadcast
+        let outbound_tx_check = OutboundTransaction::new(
+            tx_id,
+            self.dest_address.clone(),
+            self.amount,
+            MicroMinotari::zero(), // This does not matter for the check
+            sender_protocol.clone(),
+            TransactionStatus::Pending, // This does not matter for the check
+            self.message.clone(),
+            Utc::now().naive_utc(),
+            true, // This does not matter for the check
+        );
+
         // Attempt to send the initial transaction
         let SendResult {
             direct_send_result,
             store_and_forward_send_result,
             transaction_status,
-        } = match self.send_transaction(msg).await {
-            Ok(val) => val,
+        } = match check_transaction_size(&outbound_tx_check, self.id) {
+            Ok(_) => match self.send_transaction(msg).await {
```

Comment:
Nit: Wont it be easier to read by just checking the size outize of this massive match?
Something akin to 
```
if let Err(e) = check_transaction_size(&outbound_tx_check, self.id){
handle error and exit
}
```